woocommerce/woocommerce-ios
# Product Category: Support deletion

Part of #10634 


## Description

This PR updates the category list to support deleting items. Changes include: 
- Added the option to delete product categories by adding a swipe action to the category list.
- Added an alert to confirm deletion when the delete  button is tapped.
- Integrated the API to delete product categories:
  - Updated `ProductCategoriesRemote` to add a missing param `force` when deleting categories since this is required by the API.
  - Handled optimistic deletion by hiding the deleted item immediately after the user confirms the deletion to avoid having to add a progress view.
  - Handled any error by showing an alert when necessary. 

## Testing instructions

- Log in to the app and navigate to the Products tab.
- Select an existing product or create a new one if needed.
- Select the category row on the product form or select Add more details > Category if the row is not already available.
- If there are no existing categories in your store, add at least one.
- On the category list, swipe left on any row. You should see a button to delete the item.
- Tap the button, notice a confirm alert is displayed.
- Disconnect the Internet and confirm the deletion. Xcode console should log the confirmation: `🔵 Tracked product_category_settings_delete_button_tapped `.
- Notice the item is removed immediately, and after a while, an error alert is displayed.
- Reconnect the Internet and try deleting the item again.
- The item should be deleted successfully.

Please feel free to check Menu > Coupon > Edit coupon > Select categories. Deleting category should not be enabled on this screen.

## Screenshots



https://github.com/woocommerce/woocommerce-ios/assets/5533851/8794f37c-e4cc-4703-8079-09e10f66a38f




---
